rawfile/rawfile.py

```diff
@@ -22,17 +22,31 @@
 import os
 
 
+def __create_and_check_directory(dir: Path):
+    if not dir.exists():
+        logger.info("Creating directory", path=str(dir))
+        dir.mkdir(parents=True, exist_ok=True)
+    if not dir.is_dir():
+        raise RuntimeError(f"{dir} is not a directory")
+    if not os.access(dir, os.W_OK | os.R_OK | os.X_OK):
+        raise RuntimeError(f"{dir} is not accessible")
+    dir.chmod(consts.D_PERMS)
+
+
 def node_driver_preflight_checks():
+    if not config.csi_driver:
+        raise RuntimeError("CSI Driver configuration is missing")
+    if not config.csi_driver.metadata_dir:
+        raise RuntimeError("Metadata directory is not set for node plugin")
     data_dir = Path(consts.DATA_DIR)
-    if not data_dir.exists():
-        logger.info("Creating data directory", path=str(data_dir))
-        data_dir.mkdir(parents=True, exist_ok=True)
-    if not data_dir.is_dir():
-        raise RuntimeError(f"{data_dir} is not a directory")
-    if not os.access(data_dir, os.W_OK | os.R_OK | os.X_OK):
-        raise RuntimeError(f"{data_dir} is not accessible")
-    data_dir.chmod(consts.D_PERMS)
+    dirs = (
+        config.csi_driver.metadata_dir,
+        data_dir,
+    )
+    for dir in dirs:
+        __create_and_check_directory(dir)
     volume_manager.migrate_all_volume_schemas()
+    volume_manager.migrate_metadata_dir()
```

Comment:
This one should be moved before migration

Reply:
You mean migrating metadata before "all_volumes_schemas" ?

Reply:
Yeah
Should migrate the directory before the schema